src/sentry/static/sentry/app/views/settings/components/dataScrubbing/modals/add.tsx

```diff
@@ -0,0 +1,17 @@
+import {t} from 'app/locale';
+
+import ModalManager from './modalManager';
+import {ProjectSlug, Rule} from '../types';
+
+class Add<T extends ProjectSlug> extends ModalManager<T> {
```

Comment:
I would try and [avoid inheritance of components](https://reactjs.org/docs/composition-vs-inheritance.html) if possible. Could you instead just have the `ModalManager` component accept some props that customize the title and a prop to handle this `getNewRules` logic?

Reply:
also possible :)